pallets/liquidate/src/lib.rs

```diff
@@ -375,14 +377,11 @@ pub mod module {
                             classify_collaterals.iter()
                         {
                             // let repay_amount = (single_collateral_total_sum_pirce / collateral_total_value) * (debt_repay_amount * close_factor);
-                            let m: Price = 100;
-                            let repay_amount = match (close_factor
-                                * single_collateral_total_sum_pirce)
-                                .checked_mul(m)
-                                .and_then(|r| r.checked_div(collateral_total_value))
-                                .and_then(|r| r.checked_mul(debt_repay_amount))
-                                .and_then(|r| r.checked_div(m))
-                                .ok_or(Error::<T>::CaculateError)
+                            let repay_amount = match (single_collateral_total_sum_pirce
+                                .saturating_mul(close_factor.into()))
+                            .checked_div(&collateral_total_value)
+                            .and_then(|r| r.checked_mul_int(debt_repay_amount))
+                            .ok_or(Error::<T>::CaculateError)
```

Comment:
I have two questions here:
- when converting `Permill` into u128, what value turns out to be? if there is a probable overflow?
- when I calculate the percentage before, I multiply by 100 and then divide by 100, because 30/100 equals 0, not 0.3 we wish. I think we need to take more concerns here.

Reply:
first: close_factor is between 0 - 1, i think there is no overflow.
second: 100 in FixedU128 will be 100*10^18( inner type is U128, and decimal is 18 )
FixedU128::saturating_from_integer(100).checked_div(&FixedU128::saturating_from_integer(100)) equal to FixedU128::saturating_from_integer(1), the inner type is u128 and the inner value is 1*10^18

Reply:
great